code/modules/shuttle/computer.dm

```diff
@@ -36,7 +36,7 @@
 				continue
 			if(!M.check_dock(S))
 				continue
-			if(M.id == "fob" && M.getDockedId() != "fob_dock" && S.id != "fob_dock") //If the FOB isn't at the main ship, the main ship is its only potential destination
+			if(M.id == "fob" && M.getDockedId() != "fob_dock" && S.id != "fob_dock" && (S.z == ZLEVEL_STATION || S.z in SSstarmap.current_planet.z_levels)) //If the FOB isn't at the main ship, the main ship is its only potential destination. Check the z level too.

```

Comment:
split this up a bit for readability's sake. It's alright to have multiple if checks for continue.